assets/js/admin/orders/order-details/address.js

```diff
@@ -11,6 +12,88 @@ let CUSTOMER_SEARCH_RESULTS = {};
 
 jQueryReady( () => {
 
+	/**
+	 * Adjusts Overview tax configuration when the Customer's address changes.
+	 *
+	 * @since 3.0
+	 */
+	( () => {
+		const { state: overviewState } = OrderOverview.options;
+
+		if ( false === overviewState.get( 'isAdding' ) ) {
+			return;
+		}
+
+		const countryInput = document.getElementById(
+			'edd_order_address_country'
+		);
+		const regionInput = document.getElementById(
+			'edd_order_address_region'
+		);
+
+		if ( ! ( countryInput && regionInput ) ) {
+			return;
+		}
+
+		/**
+		 * Retrieves a tax rate based on the currently selected Address.
+		 *
+		 * @since 3.0
+		 */
+		function getTaxRate() {
+			const country = $( '#edd_order_address_country' ).val();
+			const region = $( '#edd_order_address_region' ).val();
+
+			const nonce = document.getElementById( 'edd_get_tax_rate_nonce' )
+				.value;
+
+			wp.ajax.send( 'edd_get_tax_rate', {
+				data: {
+					nonce,
+					country,
+					region,
+				},
+				/**
+				 * Updates the Overview's tax configuration on successful retrieval.
+				 *
+				 * @since 3.0
+				 *
+				 * @param {Object} response AJAX response.
+				 */
+				success( response ) {
+					let { tax_rate: rate } = response;
+
+					// Make a percentage.
+					rate = rate * 100;
+
+					overviewState.set( 'hasTax', {
+						country,
+						region,
+						rate,
+					} );
+				},
+				/*
+				 * Updates the Overview's tax configuration on failed retrieval.
+				 *
+				 * @since 3.0
+				 */
+				error() {
+					overviewState.set( 'hasTax', false );
+				},
+			} );
+		}
+
+		// Update rate on Address change.
+		//
+		// Wait for Region field to be replaced when Country changes.
+		// Wait for typing when Regino field changes.
+		// jQuery listeners for Chosen compatibility.
+		$( '#edd_order_address_country' ).on( 'change', _.debounce( getTaxRate, 250 ) );
+
+		$( '#edd-order-address' ).on( 'change', '#edd_order_address_region', getTaxRate );
```

Comment:
I created an order and selected a customer. I selected a customer's Tennessee address and I got this notification:

> The tax rate has been updated to 0.00%. Existing automatically calculated amounts have not been updated.

The tax rate for TN, saved in settings, is 9.25%.

If I start out by picking a different address for the customer (different state), and then change it, the tax rate is not updated.

If taxes are not enabled for the store, I think I should not see that notice, but I currently do.